File: _121_best_time_to_buy_and_sell_stock/main.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        // With Neetcode video solution
        int left = 0;
        int right = 1;
        int maxProfit = 0;

        while (right < prices.size()) {
            if (prices[left] >= prices[right]) {
                left = right;
            } else {
                maxProfit = max(maxProfit, prices[right] - prices[left]);
            }

            right++;
        }

        return maxProfit;
    }
};

File: _121_best_time_to_buy_and_sell_stock/main.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        right = 0
        profit = 0

        while right < len(prices):
            if prices[left] >= prices[right]:
                left = right
            else:
                profit = max(profit, prices[right] - prices[left])

            right += 1

        return profit